rootfs/etc/nginx/lua/balancer/sticky.lua

```diff
@@ -43,25 +45,27 @@ local function set_cookie(self, value)
     ngx.log(ngx.ERR, err)
   end
 
-  local cookie_path = self.cookie_path
+  local cookie_path = self.cookie_session_affinity.path
   if not cookie_path then
     cookie_path = ngx.var.location_path
   end
 
+  local cookie_name = self.cookie_session_affinity.name or default_cookie_name
```

Comment:
You could define a local function `cookie_name` to avoid duplicate `or`ing

Reply:
fixed. I defined a method so that`sticky_test.lua` can also use it to check that the logic is correct.

Reply:
Usually we define local function in these cases and then conditionally export it for testing if `_TEST` global variable is defined. But it's OK.